test/Makefile

```diff
@@ -9,18 +9,29 @@ GMOCK_DIR=$(LIBDIR)/googletest/googlemock
 
 # TODO: Factor out the common variables.
 CPP_NETLIB_DIR=$(LIBDIR)/cpp-netlib
+CPP_NETLIB_LIBDIR=$(CPP_NETLIB_DIR)/libs/network/src
+NETWORK_URI_DIR=$(CPP_NETLIB_DIR)/deps/uri
+NETWORK_URI_LIBDIR=$(NETWORK_URI_DIR)/src
 YAML_CPP_DIR=$(LIBDIR)/yaml-cpp
 YAML_CPP_LIBDIR=$(YAML_CPP_DIR)
 
+CPP_NETLIB_LIBS=\
+    $(CPP_NETLIB_LIBDIR)/libcppnetlib-client-connections.a \
+    $(CPP_NETLIB_LIBDIR)/libcppnetlib-server-parsers.a \
+    $(NETWORK_URI_LIBDIR)/libnetwork-uri.a
 YAML_CPP_LIBS=$(YAML_CPP_LIBDIR)/libyaml-cpp.a
 
 CPPFLAGS+= \
+    -DENABLE_KUBERNETES_METADATA \
     -isystem $(GTEST_DIR)/include -I$(GMOCK_DIR)/include \
     -I$(YAML_CPP_DIR)/include
 CXXFLAGS=\
     -std=c++11 -g -pthread -Wno-write-strings -Wno-deprecated
-LDLIBS=-lpthread -lboost_program_options -lyaml-cpp -lyajl
-LDFLAGS=-L$(YAML_CPP_LIBDIR)
+LDLIBS=\
+    -lcppnetlib-client-connections -lcppnetlib-server-parsers -lnetwork-uri \
```

Comment:
Let's make the order of the libraries match the one in `src/Makefile` exactly (since the set is now identical).